core/contextmenu_items.js

```diff
@@ -104,10 +104,28 @@ Blockly.ContextMenuItems.registerCleanup = function() {
 Blockly.ContextMenuItems.toggleOption_ = function(shouldCollapse, topBlocks) {
   var DELAY = 10;
   var ms = 0;
+  var timeouts = [];
+  var currTimeoutIdx = 0;
+  var createTimeoutFn = function(collapseFn, timerIdx) {
+    var timeoutFn = function() {
+      collapseFn();
+      timeouts.splice(timerIdx, 1, null);
+      for (var i = 0; i < timeouts.length; i++) {
+        if (timeouts[i] != null) {
+          return;
+        }
+      }
+      Blockly.Events.setGroup(false);
```

Comment:
Rather than using the state of the `timeouts` array to track whether all the `setCollapsed` calls have completed, this could be simplified by using a counter.
The counter would be incremented before the` setTimeout` call and decremented within `createTimeoutFn`. When the counter is 0, then all the `setCollapsed` calls have been made. This would remove the need to also store an array and simplify some of the checks.

Reply:
That makes much more sense, thank you @moniika. I will add these changes and update the PR.